mvg/plotting.py

```diff
@@ -294,17 +312,16 @@ def _plot_row(row_data, is_uncert_data, y_pos=0):
 
     # Modify ticks position and create legend
     df_changes = data.iloc[tick_index]
-    if time_format is None:
-        tick_x_labels = df_changes["Date"].apply(lambda x: x.date())
-    else:
-        tick_x_labels = df_changes["Date"].apply(lambda x: x.strftime(time_format))
+    tick_x_labels = df_changes["Date"].apply(
+        lambda x: x.date() if time_format is None else x.strftime(time_format)
+    )
 
     axes.set_xticklabels(tick_x_labels, rotation=timetick_angle)
     legend_labels = [
         patches.Patch(facecolor=colors[i], edgecolor="black", label="No data")
         if i == -1
         else patches.Patch(color=colors[i], label=f"Mode {int(i)}")
-        for i in list(data["labels"].unique())
+        for i in list(data["labels"].dropna().unique())
```

Comment:
When do we encounter NaN in labels? I guess this has become necessary because in the fake last row that we have added we set the `label` to `None`. I think it s better to fix this in that place rather than adding a call to fix something that is due to our augmentation of the data. I suggest we use the label from the last timestamp as the label for this extra timestamp.

Reply:
Yes, I now realize assigning `label` to `None` has become unnecessary. Your suggestion works btw.